activerecord/lib/active_record/railties/databases.rake

```diff
@@ -380,8 +380,15 @@ db_namespace = namespace :db do
     desc "Creates a db/schema.rb file that is portable against any DB supported by Active Record"
     task dump: :load_config do
       ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env).each do |db_config|
+        temp_file = Tempfile.new("temp_schema_dump.rb")
         ActiveRecord::Base.establish_connection(db_config.config)
-        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, :ruby, db_config.spec_name)
+        ActiveRecord::SchemaDumper.dump(ActiveRecord::Base.connection, temp_file)
```

Comment:
Why is not this code inside `ActiveRecord::Tasks::DatabaseTasks.dump_schema`?

Reply:
no particular reason, it was just the first place outside of the schema dumper I thought of, but the I agree the `dump_schema` method seems like more natural place, so I've moved it there.